### PR TITLE
Cross File Copy Paste

### PR DESCRIPTION
**Pull request by PJEstrada**
Add a Vuex clipboard to support copy/paste across files.

### CODE REVIEW

**Comment by PJEstrada** [comment]
Hey @anthony-sarkis,

This branch should be ready now :) Main change was that the clipboard concept was moved to vuex, this gives us the ability to paste the instances on any file without modifying the logic that much.

Also re-added the "copy all" button on the toolbar so users can copy all the instances in the file, this now uses the new.

Let me know if you have any comments or questions on the code, otherwise we can merge this to master

File: frontend/src/store.js [hunk 1]
```diff
@@ -597,6 +597,25 @@ const builder_or_trainer = {
...
+  getters:{
+    get_clipboard: state => {
+      return state.clipboard_data;
+    }
+  },
+  mutations: {
+    set_clipboard(state, data) {
+      state.clipboard_data = data
+    },
+    clear_clipboard(state, data) {
```
Comment (anthony-sarkis) [inline frontend/src/store.js:613]: do we want to call this on logout or somewhere? https://github.com/diffgram/diffgram/blob/479cfb02a07b483c477a1cd993f18b216f4b2f63/frontend/src/store.js#L97

maybe clear sets back to undefined and doesn't require data?

File: frontend/src/components/annotation/annotation_core.vue [hunk 2]
```diff
@@ -6781,19 +6807,37 @@
...
+              continue
+            }
+            let instance_clipboard = this.duplicate_instance(instance);
+            instance_clipboard.selected = false;
+            instance_clipboard.original_frame_number = this.current_frame;
+            new_instance_list.push(instance_clipboard)
+
+          }
+          this.set_clipboard(new_instance_list);
+          this.show_snackbar('All files copied into clipboard.')
```
Comment (anthony-sarkis) [inline frontend/src/components/annotation/annotation_core.vue:6823]: maybe say `instances` not `files` here?

File: frontend/src/components/annotation/annotation_core.vue [hunk 3]
```diff
@@ -26,6 +26,12 @@
                    :label_list="label_list"
                    :draw_mode="draw_mode"
                    :label_file_colour_map="label_file_colour_map"
+                   :full_file_loading="full_file_loading"
```
Comment (anthony-sarkis) [inline frontend/src/components/annotation/annotation_core.vue:29]: good re-order looks cleaner!

File: frontend/src/components/annotation/annotation_core.vue [hunk 4]
```diff
@@ -6643,66 +6648,63 @@
             console.error(error);
           }
         },
-        paste_instance: async function(next_frames = undefined, instance_hover_index = undefined){
-          if(!this.instance_clipboard && instance_hover_index == undefined){return}
-          if(instance_hover_index != undefined){
-            this.copy_instance(false, instance_hover_index)
+        add_pasted_instance_to_instance_list: async function(instance_clipboard, next_frames, original_file_id){
```
Comment (anthony-sarkis) [inline frontend/src/components/annotation/annotation_core.vue:6651]: cool reuse of this function!

**Review by anthony-sarkis — COMMENTED** [review]
Great work Pablo!!! 
I saw 2 small things to consider looking at

File: frontend/src/components/annotation/annotation_core.vue (hunk 2 above)
Comment (PJEstrada) [reply frontend/src/components/annotation/annotation_core.vue]: Thanks Anthony! Fixed here: https://github.com/diffgram/diffgram/pull/243/commits/612d7b1c577605558ea735c3c943b7dd8d36721a

File: frontend/src/components/annotation/annotation_core.vue (hunk 3 above)
Comment (PJEstrada) [reply frontend/src/components/annotation/annotation_core.vue:29]: Cool thanks Anthony!

File: frontend/src/store.js (hunk 1 above)
Comment (PJEstrada) [reply frontend/src/store.js]: Good idea! Added it here: https://github.com/diffgram/diffgram/pull/243/commits/90c325905c9d8fcadb81ed113691826b27bf6d9b

**Pull request merged**